qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -177,6 +180,30 @@ public void test70BindMountCustomPathConfAndJvmOptions() throws Exception {
         assertThat(nodesResponse, containsString("\"using_compressed_ordinary_object_pointers\":\"false\""));
     }
 
+    /**
+     * Check that the default config can be overridden using a bind mount, and that env vars are respected
+     */
+    public void test71BindMountCustomPathWithDifferentUID() throws Exception {
+        copyFromContainer(installation.config("elasticsearch.yml"), tempDir.resolve("elasticsearch.yml"));
+
+        final Path tempDataDir = mktempDir(getTempDir());
+        final Path tempEsDataDir = tempDataDir.resolve("esDataDir");
+        // Make the local directory and contents accessible when bind-mounted
+        mkDirWithPrivilegeEscalation(tempEsDataDir, 1500, 0);
+
+        // Restart the container
+        final Map<Path, Path> volumes = Map.of(tempEsDataDir.toAbsolutePath(), Path.of("/usr/share/elasticsearch/data"));
+
+        runContainer(distribution(), volumes, Map.of("ES_JAVA_OPTS", "-Xms512m -Xmx512m"));
```

Comment:
Do we need the extra env vars here? I don't see any assertions that the values are in effect. You can pass `null` otherwise.

Reply:
This is only because I wanted to keep the memory usage to a minimum for this type of test. Default is 1GB which I thought is unnecessary.

Reply:
None of the Docker tests do anything exciting right now, or spin up multiple nodes, so we could make this the default. I think the tests should all work the same way though, unless they need to change a setting for the sake of the test.

Reply:
I opted for not setting the heap in this test: 289f776bcbf0b4ce122614595c7d64883609707f
We can reduce the footprint for all tests, by default, in a future PR.